source/data-formats/modeling-bson-data.txt

```diff
@@ -11,140 +11,198 @@ Modeling BSON Data
-``document`` keys are used to distinguish the top-level BSON document from
-embedded documents, respectively.
+Sample Data
+~~~~~~~~~~~
 
-A type map may specify any class that implements
-:php:`MongoDB\BSON\Unserializable <mongodb-bson-unserializable>` as well as
-``"array"``, ``"stdClass``", and ``"object"`` (``"stdClass``" and ``"object"``
-are aliases of one another).
+The code examples in this guide reference the following sample BSON document:
 
-.. seealso::
+.. code-block:: js
 
-   :php:`Deserialization from BSON <manual/en/mongodb.persistence.deserialization.php>` in the PHP manual
+   {
+       "address" : {
+           "street" : "Pizza St", 
+           "zipcode" : "10003" 
+       },
+       "coord" : [-73.982419, 41.579505]
+       "cuisine" : "Pizza",
+       "name" : "Mongo's Pizza"
+   }
 
+Create a BSON Document
+----------------------
 
-Persistable Classes
--------------------
+You can create a BSON document by using the same notation you use to create an
+associative array in {+language+}. The {+library-short+} automatically converts
+these values into BSON documents when inserting them into a collection.
 
-The extension's :php:`persistence specification <mongodb.persistence>` outlines
-how classes implementing its :php:`MongoDB\BSON\Persistable
-<mongodb-bson-persistable>` interface are serialized to and deserialized from
-BSON. The :php:`Persistable <mongodb-bson-persistable>` interface is analogous
-to PHP's :php:`Serializable interface <class.serializable>`.
+The following example creates a BSON document that represents the preceding
+:ref:`sample BSON document <php-bson-sample>`:
 
-The extension automatically handles serialization and deserialization for
-classes implementing the :php:`Persistable <mongodb-bson-persistable>` interface
-without requiring the use of the ``typeMap`` option. This is done by encoding
-the name of the PHP class in a special property within the BSON document.
+.. literalinclude:: /includes/bson.php
+    :language: php
+    :dedent:
+    :start-after: start-create-doc
+    :end-before: end-create-doc
 
-.. note::
+Change a BSON Document
+----------------------
 
-   When deserializing a PHP variable from BSON, the encoded class name of a
-   :php:`Persistable <mongodb-bson-persistable>` object will override any class
-   specified in the type map, but it will not override ``"array"`` and
-   ``"stdClass"`` or ``"object"``. This is discussed in the
-   :php:`persistence specification <mongodb.persistence>` but it bears
-   repeating.
+You can modify the contents of a BSON document by using the same notation you use to
+modify an associative array in {+language+}. This example makes the following changes
+to the :ref:`sample BSON document <php-bson-sample>`:
 
-Consider the following class definition:
+- Adds a new ``restaurant_id`` field that has a value of ``12345``
+- Sets the value of the ``name`` field to ``"Mongo's Pizza Place"``
```

Comment:
S: I think the wording is confusing here. This code example only edits the sample BSON document in memory, not in MongoDB (as the updateOne method would do). Perhaps you can make this clearer as this isnt the same as a CRUD operation

Reply:
I wasn't sure how to make that more clear from the list, but I added a note after the code example